libraries/rush-lib/src/cli/parsing/SelectionParameterSet.ts

```diff
@@ -260,6 +282,30 @@ export class SelectionParameterSet {
     return selection;
   }
 
+  /**
+   * Computes the selected subspace when the "--subspace" parameter is provided.
+   * Returns undefined if the "--subspace" parameter is not provided
+   */
+  public getTargetSubspace(): Subspace | undefined {
+    const parameterValue: string | undefined = this._subspaceParameter?.value;
+    if (!parameterValue) {
+      return undefined;
+    }
+    if (!this._rushConfiguration.subspacesFeatureEnabled) {
+      // eslint-disable-next-line no-console
+      console.log();
+      // eslint-disable-next-line no-console
+      console.log(
+        Colorize.red(
+          `The "${this._subspaceParameter?.longName}" parameter can only be passed if "subspacesEnabled" is set to true in subspaces.json.`
+        )
+      );
```

Comment:
Sorta weird that stuff may be printed to the console when someone calls a `get<something>` function.

Reply:
🤔 True.

But this is a private API only used by CLI parsing code.  And it's conceptually throwing an exception, just using the `AlreadyReportedError` formalism for better formatting. 

We could rename `getSelectedProjectsAsync()` to something like `getAndValidateProjectsAsync()` but I wouldn't bother in this PR.